packages/cms-cli/commands/ls.js

```diff
@@ -0,0 +1,131 @@
+const {
+  addPortalOptions,
+  addConfigOptions,
+  setLogLevel,
+  getPortalId,
+  addUseEnvironmentOptions,
+} = require('../lib/commonOpts');
+const { trackCommandUsage } = require('../lib/usageTracking');
+const { logDebugInfo } = require('../lib/debugInfo');
+const { validatePortal } = require('../lib/validation');
+const {
+  loadConfig,
+  validateConfig,
+  checkAndWarnGitInclusion,
+} = require('@hubspot/cms-lib');
+const { logger } = require('@hubspot/cms-lib/logger');
+const {
+  logApiErrorInstance,
+  ApiErrorContext,
+} = require('@hubspot/cms-lib/errorHandlers');
+const {
+  getDirectoryContentsByPath,
+} = require('@hubspot/cms-lib/api/fileMapper');
+const {
+  HUBSPOT_FOLDER,
+  MARKETPLACE_FOLDER,
+} = require('@hubspot/cms-lib/lib/constants');
+
+const FOLDER_DOT_EXTENSIONS = ['functions'];
+
+const loadAndValidateOptions = async options => {
+  setLogLevel(options);
+  logDebugInfo(options);
+  const { config: configPath } = options;
+  loadConfig(configPath, options);
+  checkAndWarnGitInclusion();
+
+  if (!(validateConfig() && (await validatePortal(options)))) {
+    process.exit(1);
+  }
+};
+
+exports.command = 'ls [path]';
+exports.describe = 'get remote contents of a directory';
+
+exports.handler = async options => {
+  loadAndValidateOptions(options);
+
+  const { path } = options;
+  const directoryPath = path || '/';
+  const portalId = getPortalId(options);
+
+  trackCommandUsage('ls', {}, portalId);
+
+  logger.debug(`Getting contents of ${directoryPath}`);
+
+  const contentsResp = await getDirectoryContentsByPath(
+    portalId,
+    directoryPath
+  ).catch(async e => {
+    await logApiErrorInstance(
+      portalId,
+      e,
+      new ApiErrorContext({ portalId, directoryPath })
+    );
+    process.exit();
+  });
+
+  if (contentsResp.children.length) {
+    const mappedContents = contentsResp.children.map(fileOrFolder => {
+      const splitName = fileOrFolder.split('.');
+
+      if (
+        splitName.length > 1 &&
+        FOLDER_DOT_EXTENSIONS.indexOf(splitName[1]) === -1
+      ) {
+        return fileOrFolder;
+      }
+
+      return `/${fileOrFolder}`;
```

Comment:
Is there a particular reason for prepending `/` to some items but not others? If we decide to keep this logic, one thing to keep in mind is that modules also have a postfix (e.g. `.module`.

Reply:
This was done to make it clearer which items are folders(prepended with `/`) and which are files.

Reply:
Looks like it works properly with modules
![image](https://user-images.githubusercontent.com/6472448/96049640-50bd1280-0e46-11eb-9ec8-5c445c4ea480.png)

Reply:
Isn't `mymodule.module` a folder?

Reply:
I was trying to mimic the display of the finder in DM where modules are listed more like files, but that is a good point -- it should be listed like a folder.

![image](https://user-images.githubusercontent.com/6472448/96138364-b4d5ea00-0ecb-11eb-8dd6-87eedae60b5b.png)